### PR TITLE
Update Package.appxmanifest to have PhonePublisherId

### PR DESCRIPTION
**Pull request by sbanni**
### Description of Change

Adding PhonePublisherId to the Package.appxmanifest file to allow publishing to the Windows Store as this is required
Start Version at 1.0.0.0 instead of 0.0.0.0

This brings the template to the same defaults as the UWP templates

### Issues Fixed

[Bug 1564079](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1564079): [MAUI Publish] Error list shows 3 errors when publishing MAUI applications to the Microsoft Store

[Bug 1558236](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1558236): Maui Publish dialog version starts at 0.0.0.0 instead of 1.0.0.0

### CODE REVIEW

**Comment by mattleibow** [comment]
@jonathanpeppers @Redth should this be a msbuild property? Or is this too specific?

**Comment by jonathanpeppers** [comment]
I would say this is pretty specific to Windows and not to make an MSBuild property yet. If it doesn't map to something on all platforms, we'd basically be inventing a new thing that needs to be documented -- and it wouldn't actually simplify anything.

**Review by mattleibow — COMMENTED** [review]
I see the template has `PhonePublisherId`, but the attribute in the xml is `PhoneProductId` - is that a typo?

**Comment by mattleibow** [comment]
Also, this makes no sense... why do winui apps need phone things? this is literally something new. Not like we have a bad value, but rather we never had a value. Is Wind0ws Phone making a comeback?

**Comment by mattleibow** [comment]
I want to get some input from the Windows people to make sure there is not some hideous bug in the store worlds.

**Comment by sbanni** [comment]
> I see the template has `PhonePublisherId`, but the attribute in the xml is `PhoneProductId` - is that a typo?

PhonePublisherId is just the name, this value replaces '$guid9$' with a generated GUID

**Comment by mattleibow** [comment]
I mean the publisher name template thing is replacing the product id attribute. Why can't the template and attribute match? Especially since is a phone publisher ID as well. In fact, why is that all 0000000? There seems to be 2 values in the manifest but 1 in the template

**Comment by sbanni** [comment]
> In fact, why is that all 0000000?

The ID is required during activation on a mobile device to not be a nil value, the publisherid apparently can be nil and not matter, also to be clear I am following the convention that has existed in the UWP templates

File: src/Templates/src/templates/maui-mobile/Platforms/Windows/Package.appxmanifest [hunk 1]
```diff
@@ -2,10 +2,13 @@
 <Package
   xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
   xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
+  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
   xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
   IgnorableNamespaces="uap rescap">
 
-  <Identity Name="maui-package-name-placeholder" Publisher="CN=User Name" Version="0.0.0.0" />
+  <Identity Name="maui-package-name-placeholder" Publisher="CN=User Name" Version="1.0.0.0" />
```
Comment (mattleibow) [inline src/Templates/src/templates/maui-mobile/Platforms/Windows/Package.appxmanifest:9]: The version should be 0.0.0.0 and the build tasks will replace it with the value in the csproj. If that is not working, then that is a separate bug that needs to be fixed before we release a servicing release.

Comment (sbanni) [reply src/Templates/src/templates/maui-mobile/Platforms/Windows/Package.appxmanifest]: Leaving as 0.0.0.0

**Comment by sbanni** [comment]
@mattleibow Anything else to look at here?

**Review by mattleibow — APPROVED** [review]

**Pull request merged**